core/payment/src/service.rs

```diff
@@ -143,6 +147,58 @@ mod local {
             incoming,
         })
     }
+
+    async fn get_invoice_stats(
+        db: DbExecutor,
+        processor: PaymentProcessor,
```

Comment:
`processor` is not used here. You can remove this argument and use simple `bind` instead of `bind_with_processor`.